### PR TITLE
Remove OpenSearch feature

### PR DESCRIPTION
**Pull request by mclacore**
Closes: https://github.com/massdriver-cloud/azure-aks-cluster/issues/37

### CODE REVIEW

**Comment by WillBeebe** [comment]
Can you link a linear ticket please?

**Comment by WillBeebe** [comment]
resolves: https://linear.app/massdriver/issue/ORC-162/aks-opensearch-bug

File: massdriver.yaml [hunk 1]
```diff
@@ -230,71 +229,15 @@ params:
...
-                  required:
-                    - opensearch
-                - properties:
-                    destination:
-                      const: disabled
 
 connections:
   required:
     - azure_service_principal
-    - vnet
```
Comment (chrisghill) [inline massdriver.yaml:293]: Renaming this connection is a breaking change. Theres no way for the system to know that is a rename as opposed to a delete and recreate. This means that any existing bundles with a link between a vnet and an AKS cluster will have that link deleted.

**Review by chrisghill — CHANGES_REQUESTED** [review]

File: massdriver.yaml (hunk 1 above)
Comment (mclacore) [reply massdriver.yaml:293]: Are there currently any deployed AKS instances? If so, I'll remove this change from the PR.

File: massdriver.yaml [hunk 2]
```diff
@@ -230,71 +229,15 @@ params:
...
+    - azure_virtual_network
```
Comment (WillBeebe) [inline massdriver.yaml:236]:
```suggestion
    - vnet
```

File: massdriver.yaml [hunk 3]
```diff
@@ -294,7 +237,7 @@ connections:
   properties:
     azure_service_principal:
       $ref: massdriver/azure-service-principal
-    vnet:
+    azure_virtual_network:
```
Comment (WillBeebe) [inline massdriver.yaml:240]:
```suggestion
    vnet:
```

File: src/main.tf [hunk 4]
```diff
@@ -1,6 +1,6 @@
 resource "azurerm_resource_group" "main" {
   name     = var.md_metadata.name_prefix
-  location = var.vnet.specs.azure.region
+  location = var.azure_virtual_network.specs.azure.region
```
Comment (WillBeebe) [inline src/main.tf:3]:
```suggestion
  location = var.vnet.specs.azure.region
```

File: src/main.tf [hunk 5]
```diff
@@ -14,7 +14,7 @@ resource "azurerm_log_analytics_workspace" "main" {
 
 resource "azurerm_kubernetes_cluster" "main" {
   name                              = var.md_metadata.name_prefix
-  location                          = var.vnet.specs.azure.region
+  location                          = var.azure_virtual_network.specs.azure.region
```
Comment (WillBeebe) [inline src/main.tf:17]:
```suggestion
  location                          = var.vnet.specs.azure.region
```

File: src/main.tf [hunk 6]
```diff
@@ -40,7 +40,7 @@ resource "azurerm_kubernetes_cluster" "main" {
     vm_size             = var.node_groups.default_node_group.node_size
     min_count           = var.node_groups.default_node_group.min_size
     max_count           = var.node_groups.default_node_group.max_size
-    vnet_subnet_id      = var.vnet.data.infrastructure.default_subnet_id
+    vnet_subnet_id      = var.azure_virtual_network.data.infrastructure.default_subnet_id
```
Comment (WillBeebe) [inline src/main.tf:43]:
```suggestion
    vnet_subnet_id      = var.vnet.data.infrastructure.default_subnet_id
```

File: src/main.tf [hunk 7]
```diff
@@ -63,7 +63,7 @@ resource "azurerm_kubernetes_cluster_node_pool" "main" {
   name                  = each.value.name
   kubernetes_cluster_id = azurerm_kubernetes_cluster.main.id
   vm_size               = each.value.node_size
-  vnet_subnet_id        = var.vnet.data.infrastructure.default_subnet_id
+  vnet_subnet_id        = var.azure_virtual_network.data.infrastructure.default_subnet_id
```
Comment (WillBeebe) [inline src/main.tf:66]:
```suggestion
  vnet_subnet_id        = var.vnet.data.infrastructure.default_subnet_id
```

File: README.md [hunk 8]
```diff
@@ -196,7 +196,7 @@ Connections from other bundles that this bundle depends on.
       ```
 
   - **`specs`** *(object)*
-- **`vnet`** *(object)*: . Cannot contain additional properties.
+- **`azure_virtual_network`** *(object)*: . Cannot contain additional properties.
```
Comment (WillBeebe) [inline README.md:199]:
```suggestion
- **`vnet`** *(object)*: . Cannot contain additional properties.
```

**Review by WillBeebe — APPROVED** [review]

**Review by chrisghill — APPROVED** [review]

**Pull request merged**